nifi-extension-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/kafka/processors/ConsumeKafka.java

```diff
@@ -389,12 +474,18 @@ public void onStopped() {
         while ((service = consumerServices.poll()) != null) {
             close(service, "Processor stopped");
         }
+
+        availablePartitionedPollingContexts.clear();
+        consumerServiceToPartitionedPollingContext.clear();
     }
 
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) {
         final KafkaConsumerService consumerService = getConsumerService(context);
+        final PollingContext pollingContext = Optional.ofNullable(consumerServiceToPartitionedPollingContext.get(consumerService))
```

Comment:
Creating an `Optional` wrapper is not necessary, and does not add sufficient value for running on every invocation of `onTrigger`, so I recommend replacing with a standard conditional.